processor/transformprocessor/internal/metrics/metrics.go

```diff
@@ -295,11 +295,24 @@ func accessMetricUnit() pathGetSetter {
 func accessMetricType() pathGetSetter {
 	return pathGetSetter{
 		getter: func(ctx common.TransformContext) interface{} {
-			return ctx.(metricTransformContext).GetMetric().DataType()
-		},
-		setter: func(ctx common.TransformContext, val interface{}) {
-			if dataType, ok := val.(pmetric.MetricDataType); ok {
-				ctx.(metricTransformContext).GetMetric().SetDataType(dataType)
+			return ctx.(metricTransformContext).GetMetric().DataType().String()
+		},
+		setter: func(ctx common.TransformContext, val interface{}) {
+			if dataType, ok := val.(string); ok {
+				switch dataType {
+				case "None":
+					ctx.(metricTransformContext).GetMetric().SetDataType(pmetric.MetricDataTypeNone)
+				case "Gauge":
+					ctx.(metricTransformContext).GetMetric().SetDataType(pmetric.MetricDataTypeGauge)
+				case "Sum":
+					ctx.(metricTransformContext).GetMetric().SetDataType(pmetric.MetricDataTypeSum)
+				case "Histogram":
+					ctx.(metricTransformContext).GetMetric().SetDataType(pmetric.MetricDataTypeHistogram)
+				case "ExponentialHistogram":
+					ctx.(metricTransformContext).GetMetric().SetDataType(pmetric.MetricDataTypeExponentialHistogram)
+				case "Summary":
+					ctx.(metricTransformContext).GetMetric().SetDataType(pmetric.MetricDataTypeSummary)
+				}
```

Comment:
Based on conversation in the SIG meeting, this is not a proper way to change a metric type as it will wipe out the data.  I am going to remove the capability to set the data type for now.  I will open an issue to add the ability to change data type